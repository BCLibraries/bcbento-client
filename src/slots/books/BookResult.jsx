import React from 'react';
import VideoResult from "../video/VideoResult";
import {PrimoRecordLink} from "../PrimoRecordLink";
import AvailabilityContainer from "../availability/AvailabilityContainer";
import HathiTrustLink from "./HathiTrustLink";

function BookResult({item}) {
    if (item.isElectronic && item.type === 'Video') {
        return <VideoResult item={item} inCatalogResult={true}/>;
    }

    const link = PrimoRecordLink(item);

    return <li className="catalog-result-item media">

        <div className="media-body">
            <h3 className="catalog-result-item__media-heading media-heading">
                <a href={link} className="catalog-result-item__title">
                    {item.title}
                </a>
            </h3>

            <div className="catalog-result-item__creator">{creatorName(item)}</div>

            <div className="catalog-result-item__publisher">
                {item.publisher} {item.date}
            </div>

            <div className="catalog-result-item__type">{item.type}</div>

            {item.isElectronic && getItLink(item)}

            {item.isPhysical && physicalAvailability(item)}

            {item.hathitrustUrl && <HathiTrustLink url={item.hathitrustUrl}/> }

        </div>

        {item.coverImages.length > 1 && coverImage(item)}
    </li>
}

function creatorName(item) {
    if (item.creator) {
        return item.creator;
    }

    if (item.contributors[0]) {
        return item.contributors[0];
    }

    return '';
}

function getItLink(item) {
    return <div className="catalog-result-item__getit"><a href={PrimoRecordLink(item)}>Find online</a></div>;
}

function physicalAvailability(item) {
    return <AvailabilityContainer item={item} />;
}

function coverImage(item) {
    const altText = `Catalog record for ${item.title}`;
    return <div className="media-right">
        <a href={PrimoRecordLink(item)} aria-hidden="true">
            <img src={item.coverImages[0].url} alt={altText} className="cover-image"/>
        </a>
    </div>
}

export default BookResult;