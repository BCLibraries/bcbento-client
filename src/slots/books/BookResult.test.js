import renderer from 'react-test-renderer';
import BookResult from "./BookResult";

it('renders correctly', () => {
    const item = {
        "id": "ALMA-BC21495170660001021",
        "title": "Otters : tool users",
        "date": "[2019]",
        "type": "Book",
        "creator": "Katie Lajiness",
        "contributors": [
            "Big Buddy Books (Firm)"
        ],
        "linkableId": "ALMA-BC21495170660001021",
        "coverImages": [
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9781532115028&issn=/MC.JPG&client=primo",
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9781532115028&issn=/MC.JPG&client=primo",
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9781532115028&issn=/MC.JPG&client=primo",
            }
        ],
        "available": true,
        "availability": {
            "libraryName": "Educational Resource Center",
            "locationName": "Stacks",
            "totalCount": 1,
            "callNumber": "QL737.C25 L35 2019",
            "otherAvailabilities": false,
        },
        "holdings": [
            {
                "availabilityStatus": "available",
                "libraryCode": "ERC",
                "libraryDisplay": null,
                "locationDisplay": "Stacks",
                "callNumber": "(QL737.C25 L35 2019 )",
            }
        ],
        "hathitrustUrl": null,
        "linkToFindingAid": null,
        "isPhysical": true,
        "isElectronic": false,
        "screenCap": null,
        "mms": "99137937988901021",
    }

    const component = renderer.create(
        <BookResult item={item} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
