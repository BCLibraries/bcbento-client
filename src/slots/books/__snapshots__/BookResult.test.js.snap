// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="catalog-result-item media"
>
  <div
    className="media-body"
  >
    <div
      className="catalog-result-item__type"
    >
      Book
    </div>
    <h3
      className="catalog-result-item__media-heading media-heading"
    >
      <a
        className="catalog-result-item__title"
        href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC21495170660001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
      >
        Otters : tool users
      </a>
    </h3>
    <div
      className="catalog-result-item__creator"
    >
      Katie Lajiness
    </div>
    <div
      className="catalog-result-item__publisher"
    >
       
      [2019]
    </div>
    <ul
      className="available-items-list"
    >
      <li
        className="item-info item-info--available"
      >
        <a>
          <span
            className="item-info__library"
          >
            Find in 
            Educational Resource Center
          </span>
          <span
            className="item-info__location"
          >
            Stacks
          </span>
          <span
            className="item-info__callno"
          >
            (
            QL737.C25 L35 2019
            )
          </span>
          <span
            className="item-info__other-libraries"
          />
        </a>
      </li>
    </ul>
  </div>
  <div
    className="media-right"
  >
    <a
      aria-hidden="true"
      href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC21495170660001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
      tabIndex={-1}
    >
      <img
        alt=""
        className="cover-image undefined"
        src="https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9781532115028&issn=/MC.JPG&client=primo"
      />
    </a>
  </div>
</li>
`;
